advising/views.py

```diff
@@ -120,64 +145,89 @@ def add_minors(self, data, student):
-    """Handles retrieving SIS courses from a specific semester"""
+    """Handles retrieving timetable and SIS courses from a specific semester"""
 
-    def get(self, request, sem_name, year):
+    def get(self, request, sem_name, year, jhed=None, tt_name=None):
         """If the 'jhed' key is provided, get the courses for the student with
         the corresponding JHED. The request user must be an Advisor. Otherwise,
         get the courses for the requesting student for this semester.
 
+        If the 'tt_name' key is provided, the courses are first pulled from the
+        personal timetable with the name 'tt_name'. These courses are then
+        compared against the SIS registered sections, and if they match,
+        isVerified is True. If no key is provided, all SIS registered courses
+        will be returned with isVerified set to True for all of them.
+
         Optional data:
             jhed: The jhed of the student whose data is requested
+            tt_name: The name of the timetable to compare against
         Returns:
             registeredCourses: {
-                {**CourseSerializer(course1), is_verified: bool},
+                {**CourseSerializer(course1), isVerified: bool},
                 {...},
             }
         """
         school = request.subdomain
         semester = Semester.objects.get(name=sem_name, year=year)
-        if 'jhed' in request.data:
-            student = get_object_or_404(jhed=request.data['jhed'])
-            advisor = Student.objects.get(user=request.user)
-            transcript = get_object_or_404(
-                Transcript, owner=student, semester=semester)
-
-            if student.jhed != transcript.owner.jhed \
-                    or not advisor.is_advisor() \
-                    or advisor not in transcript.advisors.all():
+        if jhed:
+            student = get_object_or_404(Student, jhed=jhed)
+            if not self.is_advisor_for_student(request, student, semester):
                 return Response(status=status.HTTP_403_FORBIDDEN)
         else:
             student = Student.objects.get(user=request.user)
-        context = {'school': school, 'semester': semester, 'student': student}
-        courses = {'registeredCourses': []}
-        for section in student.sis_registered_courses.all():
-            course_data = {'isVerified': self.is_section_verified(
-                section, student, semester)}
 
-            courses['registeredCourses'].append(
-                dict(course_data, **CourseSerializer(
-                    section.course, context=context).data))
+        context = {'school': school, 'semester': semester, 'student': student}
+        if tt_name:
+            timetable = get_object_or_404(
+                PersonalTimetable,
+                student=student, name=tt_name,
+                school=school, semester=semester)
+            courses = self.get_registered_courses(context, timetable)
```

Comment:
Also had to add some code here that adds the courses in SIS that aren't on the timetable - but that is in my PR